core/metals.py

```diff
@@ -85,6 +85,17 @@ def on_pre_send_request_async(self, request_id: int, request: LspRequest) -> Non
                 if region.contains(point):
                     request.params['range'] = region_to_range(view, region)
 
+    def on_open_uri_async(self, uri: DocumentUri, callback: Callable[[str, str, str], None]) -> bool:
+        if uri.startswith("jar:"):
+            session = self.weaksession()
+            if not session:
+                return False
+            view = session.window.active_view()
+            view.run_command("lsp_metals_file_decoder", {"decoding_type": "jar", "file_path": uri})
```

Comment:
Since we are not using the `callback` in `on_open_uri_async`. It will not jump to the correct position after opening the file. 

Using the callback will open a scratch buffer instead. However, subsequent LSP commands didn;t seem to work with the scratch buffer.